### ISSUE TITLE
L'application ne fait pas son travail de manière continue

### ISSUE THREAD

**Issue by yassinedrissi87** [Wiqaytna-app/wiqaytna_android#29]
Au bout d'un certain moment l'application s'arrête toute seule. 
Le lancement de l'application dans les details de consommation de l'énergie est gèrée automatiquement, ce qui interdit les lancements inutiles pour économiser de l'énergie, ça pourrait être la raison derrière ces arrêts, mais j'en suis pas sûr , merci de trouver une solution à ce problème. 
![Screenshot_20200606_062802_com huawei systemmanager](https://user-images.githubusercontent.com/12575367/83936810-ea9cf080-a7be-11ea-9ae5-656c5f82bd46.jpg)

![Screenshot_20200606_062714_com huawei systemmanager](https://user-images.githubusercontent.com/12575367/83936789-cb05c800-a7be-11ea-91c4-9fb1c844bef9.jpg)

**Comment by hichamboushaba** [comment]
@yassinedrissi87 ce problème est causé par les changements de la surcouche EMUI de Huawei, et les développeurs n'ont pas beaucoup à leur disposition pour y remédier. Vous devez passer en gestion manuelle, et autoriser l'application à fonctionner en arrière-plan et à être lancée automatiquement (pour fixer ton autre souci #30).

Je pense que ce que l'équipe peut faire, c'est faire une campagne expliquant aux utilisateurs comment configurer les différents marques de smartphones pour autoriser l'éxecution en arrière plan.
Slack ont fait une chose similaire sur leur site: https://slack.com/intl/fr-fr/help/articles/360001562747-Probl%C3%A8mes-connus-avec-les-notifications-sur-Android

**Comment by IhabBendidi** [comment]
On peut voir que l'équipe doit être overloaded avec toutes les tâches liées à la gestion d'une telle appli, du coup le point fort de l'Open Source, est que nous mêmes explorons cette partie et travaillons dessus, puis faire un Pull contenant un doc expliquant les configurations des marques de Smartphone. Ce qui est préférable cependant est que @Wiqaytna définisse un code de contribution `Contribution.md` qui définit les règles de contributions, ainsi que les formats spécifiques nécessaires pour déclarer les issues. Ca pourrait aider à gérer les contributions

**Comment by hichamboushaba** [comment]
@IhabBendidi un PR ne sera pas utile pour ce point à mon avis, l'explication doit être destinée aux utilisateur finaux, donc doit être faite par le département communication/marketing sous forme de videos ou quelque chose de similaire.

**Comment by IhabBendidi** [comment]
@hichamboushaba  Oui l'explication bien sur sera faites par les équipes marketing. Ce que je voulais dire plutot est que aussi dans la tâche de trouver les différentes configs (avant de les présenter), ca peut être aussi un sujet de recherche par les contributeurs. On parle ici de bon usage et efficacité de compétences, pour avoir des résultats rapides, au lieu de surcharger l'équipe gérant les tâches critiques, avec une grande diversité de problèmes simples.

**Comment by yassinedrissi87** [comment]
@hichamboushaba en fait effectivement j'utilise un téléphone Honor,  qui est une filiale de Huawei. J'ai voulu présenter les problèmes et les bugs dans l'application du point de vue de l'utilisateur, afin que l'équipe Wiqaytna puisse les analyser dans le but de trouver des solutions catégoriques au niveau du code source.
Mais si la configuration manuelle résoud effectivement ce problème, il y a plusieurs outils pour demander aux utilisateurs de faire cette manipulation, comme le site internet de l'application, l'interface de l'application (la page d'accueil par exemple) , Tv, réseaux sociaux etc etc

**Comment by hichamboushaba** [comment]
@IhabBendidi je suis d'accord, c'est pourquoi j'ai mis le lien vers le site Slack car il liste les procédures de la majorité des modèles.

@yassinedrissi87 oui, je comprends le but de la création du bug, et je ne suis pas contre.
Mais vu la fragmentation dans le monde Android, c'est vraiment difficile d'expliquer ça dans l'app, "activer la gestion manuelle" est une procédure spécifique aux smartphones Huawei, et spécifiquement les nouveaux modèles, les autres marques nécessitent différentes procédures (Oppo, Xiaomi, Infinix...).

**Comment by yassinedrissi87** [comment]
@hichamboushaba nous pouvons donc  se baser sur l'idée de @IhabBendidi et commencer des campagnes sur les réseaux sociaux, facebook, twitter, whatsapp etc etc pour expliquer aux utilisateurs comment faire cette  configuration sur les différents modèles, en attendant que le département marketing de l'application Wikaytna le fera à sa manière.

**Comment by IhabBendidi** [comment]
@yassinedrissi87 je voulais dire plutot de travailler sur les configurations de l'application pour chaque téléphone, et tout documenter, puis l'ajouter à ce Répository dans un PR peut etre. 

Sinon partager de telles campagnes, au délà de mon opinion sur ce sujet, je crois pas que c'est un sujet à discuter dans une Issue du repo ici sur Github. On essaie de rester ici sur le plan technique de l'application préférablement et non opérationelle et marketing ou autres

**Comment by Abdelwahad** [comment]
Merci à vous pour votre intérêt et votre soutient. Vos remarques sont très pertinentes et intéressantes. Nous rejoignons l'avis de @hichamboushaba et  @yassinedrissi87 , sauf découverte d'une solution technique, c'est plutôt sur la communication/support aux utilisateurs plutôt que sur git que nous devrions prolonger l'action sur ce point. Nous allons ajouter vos recommandation de solution dans la FAQ du projets pour améliorer l'efficacité de l'application pour les utilisateurs.

**Comment by yassinedrissi87** [comment]
@Abdelwahad vous pouvez également contacter les différents fabricants mobile, pour voir avec eux ce que vous pourrez faire et ce qu'ils pourront faire à leurs niveaux pour assurer le travail de l'application de manière continue.

**Comment by aeramli** [comment]
Je ne crois pas qu'il puisse faire quoique ce soit avec les fabricants, il y a une issue sur le bugtracker de Google, beaucoup de grandes enseignes ont le même souci.
J'essayerai de partager le lien si je le retrouve.

**Comment by yassinedrissi87** [comment]
@Abdelwahad @aeramli  je viens d'installer truecaller aujourd'hui et il se met en lancement manuel par défaut, donc truecaller a la solution technique que vous cherchez, vous pouvez faire comme truecaller pour forcer l'application Wiqaytna à se mettre en lancement manuel.
